datalad/distribution/drop.py

```diff
@@ -178,53 +180,56 @@ def __call__(
                 "insufficient information for `drop`: requires at least a path or dataset")
         refds_path = Interface.get_refds_path(dataset)
         res_kwargs = dict(action='drop', logger=lgr, refds=refds_path)
+        # this try-except dance is only to maintain a previous behavior of `drop`
+        # where it did not ValueError, but yielded error status
+        try:
+            ds = require_dataset(
+                dataset, check_installed=True, purpose='dropping content')
+        except ValueError as e:
+            yield dict(
+                status='error',
+                message=str(e),
+                **res_kwargs,
+            )
+            return
+
         if dataset and not path:
             # act on the whole dataset if nothing else was specified
             path = refds_path
-        to_drop = []
-        for ap in AnnotatePaths.__call__(
-                dataset=refds_path,
+        content_by_ds = {}
+        for st in Status.__call__(
+                # no not use `ds` to preserve path semantics
+                dataset=dataset,
                 path=path,
+                annex=None,
+                untracked='no',
                 recursive=recursive,
                 recursion_limit=recursion_limit,
-                action='drop',
-                # justification for status:
-                # content need not be dropped where there is none
-                unavailable_path_status='notneeded',
-                nondataset_path_status='error',
+                eval_subdataset_state='no',
+                report_filetype='raw',
                 return_type='generator',
-                on_failure='ignore'):
```

Comment:
Should `drop` behave the same as `save` here?  It passes `on_failure='ignore'` to `status` and then lets the outer layer decide how to handle non-ok results:

https://github.com/datalad/datalad/blob/a55baf68cbcd624b467b25757bb9689a438e15d4/datalad/core/local/save.py#L208-L216

Reply:
Oh, indeed! Thanks for catching that.